src/components/views/DownloadModModal.vue

```diff
@@ -15,56 +15,14 @@
         </div>
-        <ModalCard :is-active="isOpen" :can-close="true" v-if="thunderstoreMod !== null" @close-modal="closeModal()">
-            <template v-slot:header>
-                <h2 class='modal-title' v-if="thunderstoreMod !== null">
-                    Select a version of {{thunderstoreMod.getName()}} to download
-                </h2>
-            </template>
-            <template v-slot:body>
-                <p>It's recommended to select the latest version of all mods.</p>
-                <p>Using outdated versions may cause problems.</p>
-                <br/>
-                <div class="columns is-vcentered">
-                    <template v-if="currentVersion !== null">
-                        <div class="column is-narrow">
-                            <select class="select" disabled="true">
-                                <option selected>
-                                    {{currentVersion}}
-                                </option>
-                            </select>
-                        </div>
-                        <div class="column is-narrow">
-                            <span class="margin-right margin-right--half-width"><span class="margin-right margin-right--half-width"/> <i class='fas fa-long-arrow-alt-right'></i></span>
-                        </div>
-                    </template>
-                    <div class="column is-narrow">
-                        <select class='select' v-model='selectedVersion'>
-                            <option v-for='(value, index) in versionNumbers' :key='index' v-bind:value='value'>
-                                {{value}}
-                            </option>
-                        </select>
-                    </div>
-                    <div class="column is-narrow">
-                        <span class="tag is-dark" v-if='selectedVersion === null'>
-                            You need to select a version
-                        </span>
-                        <span class="tag is-success" v-else-if='recommendedVersion === selectedVersion'>
-                            {{selectedVersion}} is the recommended version
-                        </span>
-                        <span class="tag is-success" v-else-if='versionNumbers[0] === selectedVersion'>
-                            {{selectedVersion}} is the latest version
-                        </span>
-                        <span class="tag is-danger" v-else-if='versionNumbers[0] !== selectedVersion'>
-                            {{selectedVersion}} is an outdated version
-                        </span>
-                    </div>
-                </div>
-            </template>
-            <template v-slot:footer>
-                <button class="button is-info" @click="downloadThunderstoreMod()">Download with dependencies</button>
-            </template>
-        </ModalCard>
+        <DownloadModVersionSelectModal
```

Comment:
To clarify, when we discussed separating the modals, I assumed we we're talking about making the components independent, whereas the version select modal is now a subcomponent. But I consider the subcomponent a step forwards, so it's fine for this PR. Just keep in mind we might (or might not) want to separate these even more in the future.